app/src/main/java/com/github/polypoly/app/models/game/GameViewModel.kt

```diff
@@ -43,10 +63,160 @@ class GameViewModel(
+        return remoteDB.getValue<Game>(gameUpdated.key).thenCompose { gameFound ->
+            if (gameFound.currentRound < gameUpdated.currentRound) {
+                gameData.value = gameUpdated
+                remoteDB.setValue(gameUpdated)
+            } else { // up to date version already on live db
+                gameData.value = gameFound
+                CompletableFuture.completedFuture(true)
+            }
+        }
+    }
+
+    private fun playerStateFSMTransition(expectedFrom: PlayerState, to: PlayerState) {
+        if (playerStateData.value != expectedFrom)
+            return
+        playerStateData.value = to
+    }
+
+    /**
+     * Ends ROLLING_DICE state and moves to MOVING
+     */
+    fun diceRolled() {
+        playerStateFSMTransition(PlayerState.ROLLING_DICE, PlayerState.MOVING)
+    }
+
+    /**
+     * Ends MOVING state and moves to INTERACTING
+     */
+    fun locationReached() {
+        playerStateFSMTransition(PlayerState.MOVING, PlayerState.INTERACTING)
+    }
+
+    /**
+     * Ends INTERACTING state and moves to BETTING
+     */
+    fun startBetting() {
+        playerStateFSMTransition(PlayerState.INTERACTING, PlayerState.BETTING)
+    }
+
+    /**
+     * Ends BETTING state and moves back to INTERACTING
+     */
+    fun cancelBetting() {
+        playerStateFSMTransition(PlayerState.BETTING, PlayerState.INTERACTING)
+    }
+
+    /**
+     * Resets player state back to the beginning of a turn (ie ROLLING_DICE)
+     */
+    fun resetTurnState() {
+        playerStateData.value = PlayerState.ROLLING_DICE
+    }
+
+    /**
+     * Computes the closest location to the given position among the existing locations in the game.
+     * @param position Reference position
+     * @return A future holding the closest location found or null if the closest location is farther than MAX_INTERACT_DISTANCE
+     */
+    fun computeClosestLocation(position: GeoPoint): CompletableFuture<LocationProperty?> {
+        val result = CompletableFuture<LocationProperty?>()
+
+        viewModelScope.launch {
+            var closestLocation: LocationProperty? = null
+            var closestDistance = Double.MAX_VALUE
+
+            val allLocations = gameData.value?.allLocations ?: listOf()
+            for (location in allLocations) {
+                val distance = position.distanceToAsDouble(location.position())
+                if (distance < closestDistance) {
+                    closestLocation = location
+                    closestDistance = distance
+                }
+            }
+
+            if (closestDistance > MAX_INTERACT_DISTANCE) {
+                closestLocation = null
+            }
+
+            result.complete(closestLocation)
+        }
+
+        return result
+    }
+
+    /**
+     * Rolls the dice and returns the location that corresponds to the sum of 2 dice rolls, 3 times
+     * ensuring that the player does not visit the same location twice.
```

Comment:
You just forgot to specify what the param `currentLocation` was here in the comment!